pkg/azure/aks/aks.go

```diff
@@ -88,10 +132,115 @@ func (c *Collector) CollectMetrics(_ chan<- prometheus.Metric) float64 {
 	return 0
 }
 
+func (c *Collector) populateMachinePricesCache() error {
+	c.cacheLock.Lock()
+	defer c.cacheLock.Unlock()
+	c.logger.Info("populating machine prices cache")
+
+	c.MachineStore.machineMapLock.RLock()
+	defer c.MachineStore.machineMapLock.RUnlock()
+
+	for vmName, vmInfo := range c.MachineStore.MachineMap {
+		price, err := c.PriceStore.getPriceInfoFromVmInfo(vmInfo)
+		if err != nil {
+			return err
+		}
+
+		c.MachinePricesCache[vmName] = price
+	}
+
+	return nil
+}
+
+// TODO - BREAK INTO CPU AND RAM
+func (c *Collector) getMachinePrices(vmName string) (float64, error) {
+	// Cache Hit
+	c.cacheLock.RLock()
+	if price, ok := c.MachinePricesCache[vmName]; ok {
+		c.cacheLock.RUnlock()
```

Comment:
Any reason not to defer the `c.cacheLock.RUnlock()` call? I know there's a small performance penalty, but seems like it would read a bit cleaner.